packages/dds/tree/src/test/rebaserAxiomaticTests.ts

```diff
@@ -380,6 +654,196 @@ function verifyComposeAssociativity<TChangeset>(
 	assertDeepEqual(rightPartialCompositions.at(-1), singlyComposed);
 }
 
+function verifyComposeWithInverseEqualsEmpty<TChangeset>(
+	edits: TaggedChange<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const metadata = defaultRevisionMetadataFromChanges(edits);
+
+	for (const edit of edits) {
+		const change = tagChange(edit.change, edit.revision);
+		const changeset = fieldRebaser.compose(
+			change,
+			tagChange(fieldRebaser.invert(change), change.revision),
+			metadata,
+		);
+		const actualChange = makeAnonChange(changeset);
+		assert(fieldRebaser.isEmpty !== undefined);
+		assert(fieldRebaser.isEmpty(actualChange.change));
+	}
+}
+
+function verifyComposeWithEmptyIsNoOp<TChangeset>(
+	edits: TaggedChange<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const metadata = defaultRevisionMetadataFromChanges(edits);
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	for (const edit of edits) {
+		const emptyChange = tagChange(
+			fieldRebaser.createEmpty(),
+			"test" as unknown as SessionSpaceCompressedId,
+		);
+		const changeset = fieldRebaser.compose(edit, emptyChange, metadata);
+		const actualChange = makeAnonChange(changeset);
+		assertDeepEqual(actualChange, makeAnonChange(edit.change));
+	}
+}
+
+function verifyRebaseLeftDistributivity<TChangeset>(
+	edit: TaggedChange<TChangeset>,
+	namedEditsToRebaseOver: NamedChangeset<TChangeset>[],
+	fieldRebaser: BoundFieldChangeRebaser<TChangeset>,
+) {
+	const assertDeepEqual = getDefaultedEqualityAssert(fieldRebaser);
+	assert(namedEditsToRebaseOver.length === 2, "expected at least 2 edits");
```

Comment:
We should either take exactly two base edits in the arguments or we should handle an arbitrary number of base changes. It should be pretty easy to support a list of base changes so I'm in favor of doing that.

Reply:
Updated to support an arbitrary number of base changes